More detailed descriptions of the channels (#8692)
flutter/website: src/release/upgrade.md

This command gets the most recent version of the Flutter SDK
that's available on your current Flutter channel.

If you are using the **stable** channel
and want an even more recent version of the Flutter SDK,
switch to the **beta** channel using `flutter channel beta`,
and then run `flutter upgrade`.

### Keeping informed

We publish [migration guides] for known breaking changes.

We send announcements regarding these changes to the
[Flutter announcements mailing list][flutter-announce].

To avoid being broken by future versions of Flutter,
consider submitting your tests to our [test registry].

## Switching Flutter channels

Flutter has two release channels:
**stable** and **beta**.

### The **stable** channel

> We recommend the **stable** channel
> for new users
> and for production app releases.
> It is updated roughly every three months,
> and may additionally receive occasional hot fixes
> for high-severity or high-impact issues.

Reviewer: **nits:** Write in active voice per [GSG](https://developers.google.com/style/voice). Limit adverbs as they make Google sound unsure.
```suggestion
We recommend the **stable** channel for new users
and for production app releases.
The team updates this channel about every three months.
The channel may receive occasional hot fixes
for high-severity or high-impact issues.
```

> The Flutter team's plugins and packages are continually tested against the latest **stable** release.

Reviewer: **issue:** Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
The Flutter team tests its plugins and packages against the latest **stable** release.
```

Author: We're not testing them, the computers test them. The idea is that the CI is using this branch to test the plugins. But I don't want to imply it's the only branch being used.

Author: gotcha. Changed to "The continuous integration for the Flutter team's plugins and packages includes testing against the latest stable release."

The latest documentation for the **stable** branch is at: <https://api.flutter.dev>

### The **beta** channel

> The **beta** channel has the latest stable release.
> This is the most recent version of Flutter that we have heavily tested.
> This channel has passed all our public testing,
> has been verified against test suites for Google products that use Flutter,
> and has been vetted against [contributed private test suites][test registry].
> The **beta** channel receives regular hot fixes
> to address newly discovered important issues.
> 
> The **beta** channel is essentially the same as the **stable** channel
> but updated monthly instead of quarterly.
> Indeed, when the **stable** channel is updated,
> it is updated to the latest **beta** release.

Reviewer: **issues:**
- Write in active voice per [GSG](https://developers.google.com/style/voice).
- Avoid equivocation unless needed.
```suggestion
The **beta** channel has the latest stable release.
This channel includes the most recent version of Flutter
that passed the following tests:

- All our public testing.
- Test suites for Google products that use Flutter.
- All [contributed private test suites][test registry].

The **beta** channel receives regular hot fixes
to address discovered important issues.

The **beta** channel works like the **stable** channel
but updated each month instead of each quarter.
When the **stable** channel updated,
the update comes from the latest **beta** release.
```

Author: The problem with this way of phrasing it is that it implies that there's no other tests that it passed, and I don't want to make that implication. I'm also a bit dubious about saying a channel "includes" something (the channel is just a branch and so has at any one time just one version of Flutter). Finally my concern with saying it "works like" the stable channel is that it missed the key point, which is that it really is the same as the beta channel (literally, the commit from the beta channel goes to the stable channel). Equivocation is needed there though because there are technical subtleties that mean they're not actually identical.

I'm not sure which parts are passive voice here, could you elaborate on that?

Author: > I ask "who makes the update?"

Ah, I see. I use the passive voice specifically to avoid saying who or what is performing something when I think being explicit about it will just make things more complicated in ways that are not relevant to the discussion. :-)

The answer to "who makes the update" is a whole complicated topic. It's partially automated, we're moving to a two-agent system, it's specifically not _anyone_ on the team... I'm not really sure how to succinctly say who it is who's doing it! :-)

> How is the beta channel different?

I mean, to the extent that it's different, this is what this whole document is trying to explain. What we're trying to do here is convey the idea that the only relevant difference between beta and stable is the update cadence. There are, however, lots of minor differences that are tricky to explain without getting into the weeds (I tried to explain the API docs difference and the plugin testing difference by mentioning those explicitly, but there's lots of other minor differences that are more nuanced).

"The **beta** channel contains most of the same code as the **stable** channel, save features to be released in the next quarter." isn't any more accurate (I would say it's more misleading actually).

> Is it different in that includes some patches and hotfixes?

Both beta and stable branches get cherry-picks.

> does it include new features to be released in the next stable release?

We're trying to express that "beta" branch releases are stable releases, just like releases on the "stable" branch. In fact the "stable" branch release is just a copy of a previous "beta" branch release. Thus, by definition if a feature ends up in a "stable" branch release it would have previously been in a "beta" branch release.

### Other channels

> We currently have one other channel, **master**.
> This is the channel used by people [contributing to Flutter][].
> 
> This channel is not as thoroughly tested as the **beta** and **stable** channels.
> 
> We do not recommend using this channel as it more likely to contain serious regressions.
> 
> The latest documentation for the **master** branch is at: <https://master-api.flutter.dev>

Reviewer: **issues:** Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
We have one other channel, **master**.
[Flutter contributors][] use this channel.

We test this channel less than the **beta** and **stable** channels.

We do not recommend using this channel as it might contain serious regressions.

To learn about the latest API documentation for the **master** branch,
check out <https://master-api.flutter.dev>
```

Author: I don't think "Flutter contributors" is a good link text to use for linking to the contributing guide. The guide doesn't list the people contributing. It explains how to contribute to Flutter.

I don't think "To learn about the latest API documentation" makes sense, after all, that link doesn't teach you anything about the API documentation, it _is_ the API documentation.

Author: i'm also a bit reluctant to say "we" test the channels, given that there are a number of different processes for testing that aren't all "we" (for one, most of the testing is automated).

$ flutter pub upgrade
```

> To update to the _latest possible version_ of
> all the dependencies listed in the `pubspec.yaml` file,
> and automatically update the constraints in that file,
> use the `upgrade` command with `--major-versions`:

Reviewer: **issue:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).

```suggestion
Flutter lets you update to the _latest possible version_ of
all dependencies in your `pubspec.yaml` file.
To make this update and update the constraints in that file,
run the `upgrade` command with `--major-versions`:
```

Author: "Flutter lets you" implies that we could stop you, which seems like a weird implication since this is just a text file.

Author: That would break the symmetry with the previous paragraph, which I think would harm readability, but I've made a similar change to reduce the sentence length; I put "This also automatically update the constraints in the `pubspec.yaml` file." after the command and removed the relevant clause from the paragraph in question.

$ flutter pub outdated
```

## Keeping informed

We publish breaking change announcements to the
[Flutter announcements mailing list][flutter-announce].
Aside from subscribing to receive announcements,
we'd love to hear from you!

[Flutter SDK archive]: {{site.url}}/release/archive

> [release channels]: {{site.repo.flutter}}/wiki/Flutter-build-release-channels

Reviewer 2: Looks like this link definition is still reference on line 7 - https://github.com/flutter/website/blob/a2df8ed07998abedacca1f8bfc8296ac5b482382/src/release/upgrade.md?plain=1#L7

Perhaps with this extra information, the link can be dropped?

Author: oops. yeah i am trying to remove links to the wiki from this page. The goal is for the website to be our developer documentation and the wiki the contributor documentation.

Author: fixed

Reviewer 2: I like that transition. Thanks!

> [contributing to Flutter]: https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md

Reviewer: **issue:** Change to match prior change.
```suggestion
[Flutter contributors]: https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md
```
